src/views/sms_commands/SmsCommandFormNew.js

```diff
@@ -72,6 +72,23 @@ const useResetFormFields = () => {
     )
 }
 
+// undefined = all, false = program.programType --> WITHOUT_REGISTRATION
+const shouldProgramsBeWithRegistration = parserType => {
+    // EVENT_REGISTRATION uses event programs
+    if (parserType === EVENT_REGISTRATION_PARSER.value) {
+        return false
+    }
+
+    // This is for programs of type tracker,
+    // which can have multiple program stages
+    if (parserType === PROGRAM_STAGE_DATAENTRY_PARSER.value) {
+        return true
+    }
+
+    // Unrelated yet, but in case all program stages are need in the future
+    return undefined
+}
+
```

Comment:
This seems a bit unclear to me. If I'm expecting a boolean, but undefined is also possible and meaningful, it kind of makes the function harder to understand for me. If you're dealing with multiple cases, and not just a boolean, maybe a string constant would make more sense?

Reply:
Using constants now